tests/atomic_ref/atomic_ref_is_lock_free_test.h

```diff
@@ -0,0 +1,84 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Copyright (c) 2023 The Khronos Group Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+//  Provides tests for sycl::atomic_ref is_lock_free() method
+//
+*******************************************************************************/
+#ifndef SYCL_CTS_ATOMIC_REF_IS_LOCK_FREE_TEST_H
+#define SYCL_CTS_ATOMIC_REF_IS_LOCK_FREE_TEST_H
+
+#include "atomic_ref_test_base.h"
+
+namespace atomic_ref::tests::api {
+template <typename T, typename MemoryOrderT, typename MemoryScopeT,
+          typename AddressSpaceT>
+class atomic_ref_is_lock_free_test
+    : public atomic_ref_test<T, MemoryOrderT, MemoryScopeT, AddressSpaceT> {
+  using base = atomic_ref_test<T, MemoryOrderT, MemoryScopeT, AddressSpaceT>;
+  using base::atomic_ref_test;
+
+ public:
+  void run_on_device(const std::string& type_name,
+                     const std::string& memory_order,
+                     const std::string& memory_scope,
+                     const std::string& address_space,
+                     sycl::memory_order memory_order_val =
+                         base::memory_order_for_atomic_ref_obj,
+                     sycl::memory_scope memory_scope_val =
+                         base::memory_scope_for_atomic_ref_obj) {
+    std::string description =
+        get_section_name(type_name, memory_order, memory_scope, address_space,
+                         "Check is_lock_free() method");
+    auto is_lock_free_test = [](T val_expd, T val_chgd,
+                                typename base::AtomicRT& a_r, auto result_acc,
+                                auto ref_data_acc) {
+      auto lock = a_r.is_lock_free();
```

Comment:
```suggestion
      auto is_lock_free = a_r.is_lock_free();
```
I was wondering where this lock was coming from... :-)

Reply:
Renamed)